### DISCUSSION TITLE
Is Expo WebBrowser now using ASWebAuthenticationSession?

### DISCUSSION THREAD

**Discussion by gregdburns** [expo/expo#18243] [Archived: Questions] [answered]
According to [the docs](https://docs.expo.dev/versions/latest/sdk/webbrowser/), WebBrowser is using SFAuthenticationSession, however looking at the [code](https://github.com/expo/expo/blob/main/packages/expo-web-browser/ios/WebAuthSession.swift), it appears to have been upgraded to use ASWebAuthenticationSession.

Am I looking at the wrong code, or has the module been updated to use ASWebAuthenticationSession?

If so, the docs should be updated, and this would help resolve security issues related to the prior use of SFAuthenticationSession.

Thanks!

**Comment by barthap** [comment, marked as answer]
Yes, it is using the `ASWebAuthenticationSession` 😁. The docs were wrong, there is a PR fixing them: https://github.com/expo/expo/pull/18320

**Reply by gregdburns** [reply]
Great, thanks for confirming @barthap !

**Reply by genesiscz** [reply]
Sorry to hijack so old one, but isn't it still wrong in the description up in the docs?

> expo-web-browser provides access to the system's web browser and supports handling redirects. On Android, it uses ChromeCustomTabs and on iOS, it uses SFSafariViewController or SFAuthenticationSession, depending on the method you call.
